### PR TITLE
read articles from server and added loading

### PR DESCRIPTION
**Pull request by ghost**
- articles are loaded from json-server
- created db.json file
- added loading functionality

### CODE REVIEW

File: blog_remake/script.js [hunk 1]
```diff
@@ -1,119 +1,6 @@
...
-      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Ut earum praesentium temporibus tempora? Magni, molestiae. Ab deleniti, totam a, amet pariatur doloremque iusto consequatur ducimus assumenda libero architecto id ullam dolor, ipsa voluptate laboriosam enim sunt magni magnam vitae molestiae! Soluta facilis repellendus explicabo id dolore ex nam quis nostrum.",
-    ],
-    quote:
-      "Life is like riding a bicycle, to keep your balance you must keep moving",
-    linkReadMore: "#articles/scrg2917",
-  },
-];
-
 //----------DECLARATIONS----------
+const baseUrl = " http://localhost:3000/articles";
```
Comment (alexandrumalinescu) [inline blog_remake/script.js]: you can delete "whitespace" at the beginning of const baseUrl

File: blog_remake/script.js [hunk 2]
```diff
@@ -730,8 +632,20 @@ class MyHashRouter {
 }
 
 document.addEventListener("DOMContentLoaded", () => {
-  let myRouter = new MyHashRouter();
-  const initialHash = window.location.hash;
-  window.location.hash = "#aa";
-  window.location.hash = initialHash;
+  showLoading();
```
Comment (ghost) [inline blog_remake/script.js:832]: great idea to show a loading message

**Review by ghost — APPROVED** [review]
The code is well structured. Great job.

File: blog_remake/db.json [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Saepe quaerat ratione quis impedit! Exercitationem, ipsa. Placeat quod voluptate hic nam incidunt temporibus quis blanditiis eos architecto iste commodi, est ex voluptatum veniam, ut nisi obcaecati impedit debitis! Fugiat consequatur dolorum, possimus quo magni aspernatur at quia repudiandae animi nam debitis.",
+        "Lorem ipsum dolor sit amet consectetur adipisicing elit. Ut earum praesentium temporibus tempora? Magni, molestiae. Ab deleniti, totam a, amet pariatur doloremque iusto consequatur ducimus assumenda libero architecto id ullam dolor, ipsa voluptate laboriosam enim sunt magni magnam vitae molestiae! Soluta facilis repellendus explicabo id dolore ex nam quis nostrum."
+      ],
+      "quote": "Life is like riding a bicycle, to keep your balance you must keep moving",
+      "linkReadMore": "#articles/tlod1718"
+    },
+    {
+      "id": "5",
```
Comment (ghost) [inline blog_remake/db.json]: Was it a choice to create the IDs as strings?

File: blog_remake/index.html [hunk 4]
```diff
@@ -17,6 +17,7 @@
   </head>
```
Comment (ghost) [inline blog_remake/index.html:17]: You created all elements(excepting div container) in JS, I like it. 👍

File: blog_remake/style.css [hunk 5]
```diff
@@ -316,3 +316,38 @@ input:checked + .slider:before {
...
   border-radius: 50%;
 }
+
+/*Spinner loader*/
+
+.spinner {
+  width: 40px;
+  height: 40px;
+  margin: 100px auto;
+  background-color: #333;
```
Comment (ghost) [inline blog_remake/style.css:327]: You could create root variables for all the colors you used.

File: blog_remake/script.js [hunk 6]
```diff
@@ -1,119 +1,6 @@
...
-    quote:
-      "Life is like riding a bicycle, to keep your balance you must keep moving",
-    linkReadMore: "#articles/scrg2917",
-  },
-];
-
 //----------DECLARATIONS----------
+const baseUrl = " http://localhost:3000/articles";
+let blogPosts = null;
 const navItems = [
```
Comment (ghost) [inline blog_remake/script.js:4]: I like that you structured the data as objects to be more accessible 👍💯

File: blog_remake/script.js (hunk 2 above)
Comment (ghost) [reply blog_remake/script.js:832]: Agree

File: blog_remake/script.js [hunk 7]
```diff
@@ -650,6 +537,21 @@ function addThemeControl() {
...
+  let body = document.getElementsByTagName("body")[0];
+  body.appendChild(loading);
+}
+
+function hideLoading() {
+  let container = document.getElementById("loadingContainer");
+  if (container) container.remove();
+}
+
 const currentTheme = localStorage.getItem("theme")
```
Comment (ghost) [inline blog_remake/script.js:747]: Using local storage to save the user preferences, good job!

File: blog_remake/db.json (hunk 3 above)
Comment (ghost) [reply blog_remake/db.json]: i did not realize, i think it's from prettier. I will fix this in a new commit.

File: blog_remake/style.css (hunk 5 above)
Comment (ghost) [reply blog_remake/style.css:327]: i will make a new commit with the changes 🔢

**Review by alexandrumalinescu — APPROVED** [review]

**Pull request merged**